### PR TITLE
Adding retry logic to handle node registration race conditions

### PR DESCRIPTION
**Pull request by brenton**

### CODE REVIEW

**Comment by brenton** [comment]
@detiber, PTAL.

File: roles/openshift_manage_node/tasks/main.yml [hunk 1]
```diff
@@ -1,3 +1,13 @@
+- name: Wait for Node Registration
+  delegate_to: "{{ openshift_first_master }}"
```
Comment (detiber) [inline roles/openshift_manage_node/tasks/main.yml]: delegate_to is not needed here, the the playbook containing the role is set for oo_first_master already.

**Comment by detiber** [comment]
outside of the one comment, lgtm

File: roles/openshift_manage_node/tasks/main.yml [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
+  command: >
+      {{ openshift.common.client_binary }} get node {{ item }}
+  register: omd_get_node
+  until: omd_get_node.rc == 0
+  retries: 10
+  delay: 5
+  with_items: openshift_nodes
```
Comment (dlbewley) [inline roles/openshift_manage_node/tasks/main.yml]: `openshift_nodes` evals to a large dict of facts, not a list of 2 hostnames.

Comment (detiber) [inline roles/openshift_manage_node/tasks/main.yml]: ah, indeed it does.... suggesting fix above for that.

File: playbooks/common/openshift-node/config.yml [hunk 3]
```diff
@@ -139,6 +139,8 @@
 - name: Set scheduleability
   hosts: oo_first_master
   vars:
+    openshift_nodes: "{{ hostvars
+                         | oo_select_keys(groups['oo_nodes_to_config']) }}"
```
Comment (detiber) [inline playbooks/common/openshift-node/config.yml]: per @dlbewley's comment about openshift_node being a list of facts, this should actually be:
```
    openshift_nodes: "{{ hostvars
                         | oo_select_keys(groups['oo_nodes_to_config'])
                         | oo_collect('openshift.common.hostname') }}"
```

Comment (brenton) [inline playbooks/common/openshift-node/config.yml]: Thanks! QE noticed that too.. https://bugzilla.redhat.com/show_bug.cgi?id=1241342.  I'll update it and submit it for testing.

Comment (brenton) [inline playbooks/common/openshift-node/config.yml]: I would mention that it's really odd how this even worked in my environment.

**Comment by brenton** [comment]
QE test the latest fix from @detiber and @dlbewley and it passed.  I'm going to merge this in since the issue is fairly serious and we don't want to wait any longer for install.openshift.com.

**Comment by brenton** [comment]
@twiest, FYI

**Comment by twiest** [comment]
@brenton thanks for the heads up, I like it. :+1:

**Pull request merged**